From 83bedac9fe5e84a51dc57ea21a6818729c423958 Mon Sep 17 00:00:00 2001
From: Patrik Flykt <patrik.flykt@linux.intel.com>
Date: Thu, 26 Feb 2015 13:48:56 +0200
Subject: [PATCH 11/22] service: Check whether properties can be sent before
 assigning value

Trivial logic fix.

(cherry picked from commit 50f2c9dad224c035e420bd79baed90cbfe570a80)
---
 src/service.c | 6 +++---
 1 file changed, 3 insertions(+), 3 deletions(-)

diff --git a/src/service.c b/src/service.c
index 4068283..f9ff409 100644
--- a/src/service.c
+++ b/src/service.c
@@ -4181,14 +4181,14 @@ static void set_error(struct connman_service *service,
 	if (!service->path)
 		return;
 
+	if (!allow_property_changed(service))
+		return;
+
 	str = error2string(service->error);
 
 	if (!str)
 		str = "";
 
-	if (!allow_property_changed(service))
-		return;
-
 	connman_dbus_property_changed_basic(service->path,
 				CONNMAN_SERVICE_INTERFACE, "Error",
 				DBUS_TYPE_STRING, &str);
-- 
1.9.1

